src/main/java/org/bytecodeparser/instruction/InstructionArguments.java

```diff
@@ -0,0 +1,22 @@
+package org.bytecodeparser.instruction;
+
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class InstructionArguments {
+
+    private InstructionTypes type;
```

Comment:
InstructionArguments is included in Instruction instance. InstructionTypes is included here, in InstructionArguments , and in Instruction. There is no need in such circularity, you can do either invoking ``` type.prettyPrintTypes.prettyPrintArgumentsStrategy.apply(args) ``` in Instruction prettyPrint or make prettyPrint generic in instructionArguments as it was

Reply:
Fixed. Choose first option, cause don't understand idea of second